File: app/lib/models/institution_administrator.dart
import 'institution_administrator_position.dart';

class InstitutionAdministrator {
  final String userId;
  final InstitutionAdministratorPosition position;

  const InstitutionAdministrator({
    required this.userId,
    required this.position,
  });

  factory InstitutionAdministrator.fromJson(Map<String, dynamic> json) =>
      InstitutionAdministrator(
        userId: json['userId'] as String,
        position: InstitutionAdministratorPosition.fromJson(
          json['position'] as Map<String, dynamic>,
        ),
      );
}

File: backend/internal/model/institution_administrator.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
	"time"
)

// InstitutionAdministrator is a table (model), that contains info, related only
// to institution administrators. This table extends the "users" table
type InstitutionAdministrator struct {
	// one-to-one (administrator-to-user)
	UserID uuid.UUID `gorm:"type:uuid;primaryKey"`
	User   User      `gorm:"foreignKey:UserID;references:ID"`

	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt

	PositionID uint16
	Position   InstitutionAdministratorPosition `gorm:"foreignKey:PositionID;references:ID"`
}
